x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/IdentifierGenerator.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql;
+
+import org.elasticsearch.test.ESTestCase;
+
+public class IdentifierGenerator {
+
+    /**
+     * Generate random identifier that could be used as a column name
+     */
+    public static String randomIdentifier() {
+        return ESTestCase.randomIdentifier();
+    }
+
+    /**
+     * Generates one or several coma separated index patterns
+     */
+    public static String randomIndexPatterns(Feature... features) {
+        return maybeQuote(String.join(",", ESTestCase.randomList(1, 5, () -> randomIndexPattern(features))));
+    }
+
+    /**
+     * Generates a random valid index pattern.
+     * You may force list of features to be included or excluded using the arguments, eg {@code randomIndexPattern(PATTERN, not(HIDDEN))}.
+     * Identifier could be an index or alias. It might be hidden or remote or use a pattern.
+     * See @link <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html#indices-create-api-path-params">valid index patterns</a>
+     */
+    public static String randomIndexPattern(Feature... features) {
```

Comment:
This thing is really cool. Thanks for researching all the different ways that patterns can look like!